dadi/lib/index.js

```diff
@@ -692,9 +692,8 @@ Server.prototype.addComponent = function (options, reload) {
               return options.component[req.method.toLowerCase()](req, res, next)
             })
             .catch(err => {
-              if (err) return next(err)
```

Comment:
@jimlambie as we're now passing an actual error with the `reject(new Error('...` vs reject('...` I've change this so the 404 catches the error, rather than passing a 500 and stopping the app. Happy?